modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/RocksDbMvPartitionStorage.java

```diff
@@ -1015,6 +1123,218 @@ public void destroyData(WriteBatch writeBatch) throws RocksDBException {
         writeBatch.deleteRange(cf, partitionStartPrefix(), partitionEndPrefix());
     }
 
+    @Override
+    public @Nullable TableRowAndRowId pollForVacuum(HybridTimestamp lowWatermark) {
+        return busy(() -> {
+            throwExceptionIfStorageInProgressOfRebalance(state.get(), this::createStorageInfo);
+
+            WriteBatchWithIndex batch = requireWriteBatch();
+
+            // We retrieve the first element of the GC queue and seeking for it in the data CF.
+            // However, the element that we need to garbage collect is the next (older one) element.
+            // First we check if there's anything to garbage collect. If the element is a tombstone we remove it.
+            // If the next element is exists, that should be the element that we want to garbage collect.
+            try (
+                    var gcUpperBound = new Slice(partitionEndPrefix());
+                    ReadOptions gcOpts = new ReadOptions().setIterateUpperBound(gcUpperBound).setTotalOrderSeek(true);
+                    RocksIterator gcIt = db.newIterator(gc, gcOpts);
+            ) {
+                gcIt.seek(partitionStartPrefix());
+
+                if (invalid(gcIt)) {
+                    // GC queue is empty.
+                    return null;
+                }
+
+                ByteBuffer gcKeyBuffer = allocateDirect(GC_KEY_SIZE);
+                gcKeyBuffer.clear();
+
+                ByteBuffer dataKeyBuffer = MV_KEY_BUFFER.get();
```

Comment:
Why do we clear this buffer here? What if the lowWatermark is too small and we exit early?